backend/core/src/main/java/org/sonarsource/sonarlint/core/sync/SynchronizationService.java

```diff
@@ -173,15 +178,25 @@ private void synchronizeProjectsOfTheSameConnection(String connectionId, Map<Str
     }
     serverApiProvider.getServerApi(connectionId).ifPresent(serverApi -> {
       var subProgressGap = progressGap / boundScopeBySonarProject.size();
-      var subProgress = progress;
+      var subProgress = new AtomicDouble(progress);
       for (var entry : boundScopeBySonarProject.entrySet()) {
         var sonarProjectKey = entry.getKey();
-        notifier.notify("Synchronizing project '" + sonarProjectKey + "'...", Math.round(subProgress));
-        issueSynchronizationService.syncServerIssuesForProject(connectionId, sonarProjectKey, cancelMonitor);
-        taintSynchronizationService.synchronizeTaintVulnerabilities(connectionId, sonarProjectKey, cancelMonitor);
-        hotspotSynchronizationService.syncServerHotspotsForProject(connectionId, sonarProjectKey, cancelMonitor);
-        synchronizedConfScopeIds.addAll(entry.getValue().stream().map(BoundScope::getConfigScopeId).collect(toSet()));
-        subProgress += subProgressGap;
+        var allScopes = configurationRepository.getBoundScopesToConnectionAndSonarProject(connectionId, sonarProjectKey);
+        var allScopesByOptBranch = allScopes.stream()
+          .collect(groupingBy(b -> sonarProjectBranchTrackingService.awaitEffectiveSonarProjectBranch(b.getConfigScopeId())));
+        allScopesByOptBranch
+          .forEach((branchNameOpt, scopes) -> branchNameOpt.ifPresent(branchName -> {
+            var branchBinding = new BranchBinding(new Binding(connectionId, sonarProjectKey), branchName);
+            if (shouldSynchronizeBranch(branchBinding)) {
+              branchSynchronizationTimestampRepository.setLastSynchronizationTimestampToNow(branchBinding);
+              notifier.notify("Synchronizing project '" + sonarProjectKey + "'...", (int) Math.round(subProgress.get()));
+              issueSynchronizationService.syncServerIssuesForProject(serverApi, connectionId, sonarProjectKey, branchName, cancelMonitor);
+              taintSynchronizationService.synchronizeTaintVulnerabilities(serverApi, connectionId, sonarProjectKey, branchName, cancelMonitor);
+              hotspotSynchronizationService.syncServerHotspotsForProject(serverApi, connectionId, sonarProjectKey, branchName, cancelMonitor);
+              synchronizedConfScopeIds.addAll(entry.getValue().stream().map(BoundScope::getConfigScopeId).collect(toSet()));
+              subProgress.getAndAccumulate(subProgressGap, Double::sum);
```

Comment:
I think the accumulation of subProgress should be right under the for loop that iterates the `boundScopeBySonarProject` collection, and `subProgress` was divided into its size.

Reply:
I think you're right. It was already wrong before, I will fix this